### PR TITLE
Add backend check to `validate`

### PR DESCRIPTION
**Pull request by grantjoy**
### Background

Validate only works over API key. This helps users understand that the validate mechanism doesn't work in the way they are attempting to use it.

### Changes

* Adds a better error message when validate is called without an API key being set

### Testing

* manually tested `pat validate`

### CODE REVIEW

**Review by nskobov — APPROVED** [review]

File: panther_analysis_tool/util.py [hunk 1]
```diff
@@ -140,6 +140,9 @@ def get_backend(args: argparse.Namespace) -> BackendClient:
             )
         )
 
+    if not hasattr(args, "aws_profile"):
+        return False
```
Comment (darwayne) [inline panther_analysis_tool/util.py:144]: hmm linter isn't happy about this since there's a type mismatch

Comment (stedrow) [inline panther_analysis_tool/util.py:144]:
```suggestion
        return 1, "--aws-profile wasn't set to a valid aws profile, please include a valid aws-profile in your arguments and try again."
```

Comment (grantjoy) [reply panther_analysis_tool/util.py:144]: thank you

**Comment by grantjoy** [comment]
closing in favor of #380

**Pull request closed without merging**